### DISCUSSION TITLE
Why AssistantAgent designed to not have a functions ?

### DISCUSSION THREAD

**Discussion by alexey2baranov** [microsoft/autogen#1793] [General]
Hello! 

I learning autogen a couple of days. It seems interesting but sometimes strange for me.

I would like to know why assistant agents designed to avoid functions? In notebooks I see that an assistant send a special type of message which is tool_call to a user proxy. And the user proxy has implementation of this function.  

This seems strange for me. As I understand generally tools designed to be executed at the same place as they are defined. But in Autogen they are not.  So in couple words could you expain what is the purpose of such design?

**Comment by gagb** [comment]
I think it depends on the definition of "agents". You could have an agent where tool_call and its execution are in the same agent. The contrast between the AssistantAgent and the new nested chats are good example of that. @qingyun-wu might have more thoughts on this.

**Reply by alexey2baranov** [reply]
could you elaborate on what is new nested chat please?
